### ISSUE TITLE
The latest version select default candidate word is wrong

### ISSUE THREAD

**Issue by kdurant** [Shougo/neocomplete.vim#544]
# Problems summary
use `Enter` select default candidate word is wrong.
Eg.
When you enter  `Eng`, it will pop `English`, if you use `Enter` key, it don't complete word and newline.

This error don't display with gvim verision 1-460

## Environment Information
 * OS:windows 10
 * Vim version:1-1832

## Provide a minimal .vimrc with less than 50 lines (Required!)

```vim
    let g:neocomplete#enable_insert_char_pre = 1
    let g:neocomplete#enable_at_startup = 1
    let g:neocomplete#enable_smart_case = 1
    " Set minimum syntax keyword length.
    let g:neocomplete#auto_completion_start_length = 1
    let g:neocomplete#enable_auto_select = 1
    let g:neocomplete#enable_refresh_always = 1
    let g:neocomplete#sources#syntax#min_keyword_length = 3
    let g:neocomplete#lock_buffer_name_pattern = '\*ku\*'

    if !exists('g:neocomplete#keyword_patterns')
        let g:neocomplete#keyword_patterns = {}
    endif
    let g:neocomplete#keyword_patterns['default'] = '\h\w*'
    let g:neocomplete#enable_ignore_case = 0
    "complete all buffers
    let g:neocomplete#same_filetypes = {}
    let g:neocomplete#same_filetypes._ = '_'

    inoremap <expr><C-h> neocomplete#smart_close_popup()."\<C-h>"
    inoremap <expr><BS> neocomplete#smart_close_popup()."\<C-h>"

    let g:neocomplete#data_directory='$HOME\PluginConfig\neocomplete'

    nmap     <M-b>   :NeoCompleteBufferMakeCache<cr>
    inoremap <expr><M-e>  neocomplete#cancel_popup()

    set completefunc=neocomplete#complete#auto_complete

```

## The reproduce ways from Vim starting (Required!)

 1. foo
 2. bar
 3. baz

**Comment by Shougo** [comment]
https://github.com/vim/vim/pull/875 fixes the problem.

**Comment by Shougo** [comment]
Fixed by Vim 7.4.2014.
